src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs

```diff
@@ -497,8 +504,6 @@ private static string GenerateRowField(string val, int width, int alignment, Dis
             {
                 return s;
             }
-            // we have to pad
-            System.Diagnostics.Debug.Assert(finalValueDisplayLength == width - 1, "padding is not correct");
```

Comment:
[This is not a blocking comment]
@SteveL-MSFT I just noticed that in the `switch (alignment)` block below, `addPadding` is checked only when padding for `left alignment`, not when padding for `center alignment`. This is not in the same pattern as in the `switch (alignment)` block in `if (currentValueDisplayLength < width)`. Just want to bring it up so you can take a another look to see if it's intended.

Reply:
Good catch, let me check this.  I'll add an extra test for truncation.